### ISSUE TITLE
add submitCount to useField

### ISSUE THREAD

**Issue by giladv** [jaredpalmer/formik#2207]
### Current Behavior

currently there's no way to build a field which shows the error after the field was touched or the form was submitted with the useField hook.

### Desired Behavior

```const [field, meta, helpers, form] = useField(props);```

and:
```
(meta.touched || form.submitCount) && meta.error && <div>{meta.error}</div>
```

**Comment by mrmuhammadali** [comment]
All of the registered fields get tocuhed when a form is submitted so there is no need to check for `form.submitCount`. You can fulfill your use case by dispatching `onBlur` event of the field or explicitly calling `helpers.setTouched(true)`.

**Comment by giladv** [comment]
@mrmuhammadali this is only true if you provide initialValues prop with all the fields. otherwise it won't work. very uncomfortable

**Comment by johnrom** [comment]
@giladv this is true, and unfortunately is how Formik works. Formik's API isn't currently built around `<Field />` registration, because Fields may or may not be rendered at a given time. InitialValues is described as required at the following locations: 

https://github.com/jaredpalmer/formik/issues/445#issuecomment-366952762
https://jaredpalmer.com/formik/docs/guides/form-submission#pre-submit

@jaredpalmer we might want to make this fact more prevalent in the overview and/or tutorial pages. The first paragraph where Formik props are described should clearly state that initialValues is the model which Formik's internal state is based upon. It's very clear in TypeScript but not everyone uses it.

**Comment by clementoriol** [comment]
Bumping this, I've got clients often asking for the "show errors only after 1 submit", and while this is doable by using `useFormikContext` from the field to get the submitCount, I'd prefer using getting the whole context each time, especially now that v3 optimizations are coming.

Any preferred way of doing this ?

**Comment by jfortez** [comment]
Any news on this case?

**Comment by taemoolopliam** [comment]
Feature request

**Comment by taemoolopliam** [comment]
const formik = useFormikContext();

    if (formik.submitCount > 0) {
        meta.touched = true;
    }
